### PR TITLE
rootfs: add shell-tables + fixes

### PR DESCRIPTION
**Pull request by sixtyfourktec**
Search for *.shell-table files in the .bob directories and cat them into one /etc/shells. Some login tools check for the existence of an entry for the used shell in /etc/shells. Packages which provide shells, can simply create such a file with an appropriate entry.

### CODE REVIEW

File: recipes/utils/screen.yaml [hunk 1]
```diff
@@ -1,28 +1,37 @@
-inherit: [autotools, patch]
+inherit: [autotools]
 
 metaEnvironment:
-  PKG_VERSION: 5.0.0
+    PKG_VERSION: 5.0.0
```
Comment (rhubert) [inline recipes/utils/screen.yaml:4]: Hmm...any reason for changing all the indentations? Do we have a style guide for recipes?

Comment (sixtyfourktec) [reply recipes/utils/screen.yaml:4]: I don't think there is any style guide. I just copied the formatting of the other files checked into those repositories.

File: recipes/core/busybox.yaml [hunk 2]
```diff
@@ -31,6 +31,20 @@ buildScript: |
     makeParallel
     make CONFIG_PREFIX=${BOB_CWD}/install install
 
-packageScript: |
-    cp -a $1/install/* .
+    mkdir -p install/.bob
+
+    # busybox must be setuid
+    cat >install/.bob/busybox.device-table <<EOF
```
Comment (jkloetzke) [inline recipes/core/busybox.yaml:37]: Theoretically we should test for `CONFIG_FEATURE_SUID`. Otherwise that would create a security hole. OTOH, I don't think anybody will ever disable this feature.

File: recipes/core/busybox.yaml [hunk 3]
```diff
@@ -31,6 +31,20 @@ buildScript: |
...
+    /usr/bin/busybox f 4755 0  0 - - - - -
+    EOF
 
+    # add configured shells
+    declare -a BB_SHELLS="/bin/sh /usr/bin/sh"
+    grep -qsE '^CONFIG_SHELL_ASH=y$' build/.config && BB_SHELLS+=" /bin/ash /usr/bin/ash"
+    grep -qsE '^CONFIG_SHELL_HUSH=y$' build/.config && BB_SHELLS+=" /bin/hush /usr/bin/hush"
+    cat >install/.bob/busybox.shell-table <<EOF
+    ${BB_SHELLS// /$'\n'}
+    EOF
```
Comment (jkloetzke) [inline recipes/core/busybox.yaml:47]: Even though `BB_SHELLS` is declared as array, it is not used as such. Why not just print to the file directly?

```sh
(
    printf '/bin/sh\n/usr/bin/sh\n'
    grep -qE '^CONFIG_SHELL_ASH=y$' build/.config && printf '/bin/ash\n/usr/bin/ash\n'
    grep -qE '^CONFIG_SHELL_HUSH=y$' build/.config && printf '/bin/hush\n/usr/bin/hush\n'
)  > install/.bob/busybox.shell-table
```

BTW, why the `-s` switch? Wouldn't a missing `build/.config` be an error that should be visible?

File: recipes/utils/screen.yaml (hunk 1 above)
Comment (jkloetzke) [reply recipes/utils/screen.yaml:4]: Right, there is not style guide. I guess we should add some, though. So far we mostly used 4 spaces.

But please separate style fixes from actual changes into separate commits. Conflating both makes it harder to see the actual changes.

File: recipes/net/openssh.yaml [hunk 4]
```diff
@@ -26,6 +26,13 @@ buildScript: |
         --with-ssl-dir="${BOB_DEP_PATHS[libs::openssl-dev]}/usr" \
         --without-openssl-header-check
 
+    mkdir -p install/.bob
+
+    # add some daemon users
+    cat >install/.bob/sshd.user-table <<EOF
```
Comment (jkloetzke) [inline recipes/net/openssh.yaml:32]: The version bump and the daemon user should go into dedicated commits each.

File: recipes/utils/screen.yaml (hunk 1 above)
Comment (sixtyfourktec) [reply recipes/utils/screen.yaml:4]: done

File: recipes/core/busybox.yaml (hunk 2 above)
Comment (sixtyfourktec) [reply recipes/core/busybox.yaml]: done

File: recipes/core/busybox.yaml (hunk 3 above)
Comment (sixtyfourktec) [reply recipes/core/busybox.yaml]: done

File: recipes/net/openssh.yaml (hunk 4 above)
Comment (sixtyfourktec) [reply recipes/net/openssh.yaml:32]: done

**Comment by jkloetzke** [comment]
Thanks.
